### PR TITLE
edopro: Fixups

### PR DESCRIPTION
**Pull request by OPNA2608**
- As discussed off-site, bundling an `ocgcore` causes more problems than it solves due to the auto-updating nature of the card scripts. Nuke it from the package, configure it to load downloaded solib instead.
- We never use the `ocgcore` submodule from `edopro`, so don't fetch it.
- Add `meta` information to relevant intermediary derivations. Mostly relevant for `meta.platforms` so Hydra plays along with providing prebuilt packages.
- Install the custom Irrlicht's lib to `$out/lib`.
- `xprop WM_CLASS` says the `WM_CLASS` is `edopro` in our case, so adjust the desktop file.
- Generate more size variants of the icon. Some desktops have more or less of an affinity for certain sizes than others.
- Fix the update script. `nix-prefetch-github` was erroring out about missing `nix-prefetch-git` / `nix-prefetch-url` / `nix-build`, adding `nix-prefetch-scripts` fixed it.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Redhawk18** [comment]
You've tested all of this right?

**Comment by Redhawk18** [comment]
I'm confused how this wasn't found by you on the original pr because this pr is basically reverting all the changes your requested from me in the first place.

**Comment by OPNA2608** [comment]
> You've tested all of this right?

Yes, as discussed on Discord.

> I'm confused how this wasn't found by you on the original pr

Tired, busy, not infinite amounts of time, the expectation that you would also double-check if the suggestions are indeed working properly. We're both just human, mistakes happen.

> this pr is basically reverting all the changes your requested from me in the first place.

Some reverts are part of this, as discussed on Discord & listed in the OP. I wasn't aware how necessary having an up-to-date ocgcore was, and presumably neither did you.

**Comment by Redhawk18** [comment]
I don't have an arm64 machine to have tested any of this on, thats why I relied on you.

**Comment by Redhawk18** [comment]
After that we should be fine, the core auto updates.

**Comment by Redhawk18** [comment]
It puts arm64 in a bad spot but I'm not sure what to do, we could also watch for master updates to the core?

**Comment by OPNA2608** [comment]
(putting upstream's response to some questions here)

> > would it be possible to get regular aarch64-linux supported with the ocgcore distribution method?
> 
> Atm it's only being built by the ci but not distributed because we don't provide aarch64 linux builds, since they're very low in demand I'll have to think about it, ~~mostly how to name it in case~~
> 
> > The hardcoded libocgcore.so is x86_64 so that obv doesn't work. patching it to try libocgcorev8.so also fails, I assume because it's targeting android and thus not compatible enough?
> 
> It boils down to different libc _I once tried running a musl libc core on android, but that actually didn't work, I might have to retry it, who knows maybe they can all be merged into one, in which case the distribution problem would be solved as well
> 
> > my laptop is aarch64, I'd like to play against the AI and do the puzzles on it.
> 
> Unfortunately for now your only choice is to build things manually

I don't think starting a wild goose chase to catch up with the distributed card scripts' required ocgcore version any time a new one is needed makes sense. Let's just use the distributed solibs. Sucks for me, but oh well.

**Comment by Redhawk18** [comment]
I believe you doubt the ability nixos has now a days to auto update and auto merge update pr's. But it works on my machine so I'm not too pressed

**Comment by Redhawk18** [comment]
Honestly if we just version the whole repo folder it could just be fine though, but we could revisit that. But I'm not sure because you don't want to touch user files.

**Comment by Redhawk18** [comment]
@OPNA2608 Have you had a chance to look over my suggestions?

**Comment by Redhawk18** [comment]
The current desktop file in nixpkgs, but its fixed in this pr

**Comment by Redhawk18** [comment]
> I'm unsure what suggestions you're referring to.

It's not a suggestion, I'm referring to edopro in nixpkgs unstable currently. I thought it should be noted.

**Comment by Redhawk18** [comment]
> > Have you had a chance to look over my suggestions?
> 
> > It's not a suggestion
> 
> I don't understand.

Does the .desktop file actually work? Because it does not work in nixpkgs currently.

**Comment by OPNA2608** [comment]
Elaborate on "doesn't work" please.

**Comment by Redhawk18** [comment]
> Elaborate on "work" and "does not work" please.

image

**Comment by OPNA2608** [comment]
> image

[simplescreenrecorder-2024-10-31_12.10.23.webm](https://github.com/user-attachments/assets/acdc107a-3d16-4b24-8839-0fc8b636965d)

**Comment by Redhawk18** [comment]
Just apply my suggestions and merge the pr, I don't have time for nitpicking

**Comment by OPNA2608** [comment]
What suggestion(s). There *are* no suggestion(s). That's what I keep asking about, because I don't have the time for this back & forth either.

**Comment by Redhawk18** [comment]
> What suggestion(s). There _are_ no suggestion(s). That's what I keep asking about, because I don't have the time for this back & forth either.
> 
> > > "Have you had a chance to look over my suggestions?"
> > 
> > 
> > I'm unsure what suggestions you're referring to.
> > > "It's not a suggestion, […]"

Do you not see the review above? Because I see it on my side.

**Review by Redhawk18 — CHANGES_REQUESTED** [review]

**Comment by Redhawk18** [comment]
Sorry about that it was pending, I thought that meant you were pending on making the changes not me.

File: pkgs/by-name/ed/edopro/update.py
Comment (OPNA2608) [reply pkgs/by-name/ed/edopro/update.py:2]: From the OP:

> - Fix the update script. `nix-prefetch-github` was erroring out about missing `nix-prefetch-git` / `nix-prefetch-url` / `nix-build`, adding `nix-prefetch-scripts` fixed it.

Running the script the way an automatic update would be attempted was failing:

```
$ env NIX_PATH=nixpkgs=$PWD nix-shell maintainers/scripts/update.nix --argstr package edopro

Going to be running update for following packages:
 - edopro-application-40.1.0

Press Enter key to continue...

Running update for:
 - edopro-application-40.1.0: UPDATING ...
 - edopro-application-40.1.0: ERROR
 
 --- SHOWING ERROR LOG FOR edopro-application-40.1.0 ----------------------
 
 [...]

Traceback (most recent call last):
  File "/home/puna/Development/nixpkgs/pkgs/by-name/ed/edopro/update.py", line 59, in <module>
    edopro_hash = get_hash("edo9300", "edopro", edopro.commit.sha)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/puna/Development/nixpkgs/pkgs/by-name/ed/edopro/update.py", line 54, in get_hash
    out_json = json.loads(out.stdout.decode())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/lib/python3.12/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/lib/python3.12/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/lib/python3.12/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

And printing `out.stderr.decode()` to `stderr` revealed the following error:
```
Prefetch failed: Unable to calculate hash sum. nix-prefetch-github failed to calculate a sha256 hash for the requested github repository. Do you have nix-prefetch-git, nix-prefetch-url and nix-build in your PATH?
```

But I *think* it was working fine when I tried to rebase this on master… before `githubkit` stopped working due to me running the script too much & getting hit with a `githubkit.exception.PrimaryRateLimitExceeded`.

I'll just drop this again for now. If it comes up again, it'll re-submit this in another PR.

Comment (Redhawk18) [reply pkgs/by-name/ed/edopro/update.py]: Interesting, It worked perfectly for me. But it's fine I don't really care.

**Review by Redhawk18 — APPROVED** [review]

**Comment by OPNA2608** [comment]
@wegank is the `12.approved-by: package-maintainer` label not appropriate here? @Redhawk18 co-maintains the derivations.

**Comment by wegank** [comment]
I guess I should let `label-approved` not touch the label when `ofborg-eval-check-maintainers` hasn't been run...

**Comment by Redhawk18** [comment]
Oh great Ofborg, please bless us. :pray:

**Comment by OPNA2608** [comment]
Builds on ofborg, checked once again that offline mode works. Merging.

**Pull request merged**
